common/web/keyboard-processor/src/text/keyEvent.ts

```diff
@@ -10,18 +10,21 @@ import type Keyboard from "../keyboards/keyboard.js";
 import { type DeviceSpec } from "@keymanapp/web-utils";
 
 import Codes from './codes.js';
-import DefaultRules from './defaultRules.js';
 import { ActiveKeyBase } from "../index.js";
 
+export interface DefaultRulesInterface {
+  forAny(Lkc: KeyEvent, isMnemonic: boolean): string;
+}
+
+let defaultRules: DefaultRulesInterface;
+
+export function setDefaultRules(rules: DefaultRulesInterface) {
+  defaultRules = rules;
+}
+
```

Comment:
I'm fairly certain that we can drop the need for this entirely, though it does appear a bit of work in another direction might be required.  More on this below in another comment within this file.

I can understand if we wish to defer these changes for a little while, but I think it's something that should be addressed moving forward due to requirements for mnemonic keyboard layout handling.

I'd be cool with it being done in a followup PR if needed.

Reply:
Done